sql_request_abstract/models/sql_request_mixin.py

```
@@ -0,0 +1,255 @@
+            * 'fetchall': execute the select request, and return the
+                result of 'cr.fetchall()'.
+            * 'fetchone' : execute the select request, and return the
+                result of 'cr.fetchone()'
+        :param rollback: (boolean) mention if a rollback should be played after
+            the execution of the query. Please keep this feature enabled
+            for security reason, except if necessary.
+            (Ignored if @mode in ('view', 'materialized_view'))
+        :param view_name: (str) name of the view.
+            (Ignored if @mode not in ('view', 'materialized_view'))
+        :param copy_options: (str) mentions extra options for
+            "COPY request STDOUT WITH xxx" request.
+            (Ignored if @mode != 'stdout')
+
+        ..note:: The following exceptions could be raised:
+            psycopg2.ProgrammingError: Error in the SQL Request.
+            openerp.exceptions.Warning:
+                * 'mode' is not implemented.
+                * materialized view is not supported by the Postgresql Server.
+        """
+        self.ensure_one()
+        res = False
+        # Check if the request is in a valid state
+        if self.state == 'draft':
+            raise UserError(_(
+                "It is not allowed to execute a not checked request."))
+
+        # Disable rollback if a creation of a view is asked
+        if mode in ('view', 'materialized_view'):
+            rollback = False
+
+        params = params and params or {}
+        query = self.env.cr.mogrify(self.query, params).decode('utf-8')
+
+        if mode in ('fetchone', 'fetchall'):
+            pass
+        elif mode == 'stdout':
+            query = "COPY (%s) TO STDOUT WITH %s" % (query, copy_options)
+        elif mode in 'view':
+            query = "CREATE VIEW %s AS (%s);" % (query, view_name)
+        elif mode in 'materialized_view':
+            self._check_materialized_view_available()
```

If not available, nothing will happen, since `_check_materialized_view_available()` returns a Boolean and nothing is done with it here. The check method should raise an error.